src/libcollections/ring_buf.rs

```diff
@@ -121,6 +118,19 @@ impl<T> RingBuf<T> {
             self.ptr.offset(src as int) as *const T,
             len);
     }
+
+    /// Copies a contiguous block of memory len long from src to dst
+    #[inline]
+    unsafe fn copy_nonoverlapping(&self, dst: uint, src: uint, len: uint) {
+        debug_assert!(dst + len <= self.cap, "dst={} src={} len={} cap={}", dst, src, len,
+                      self.cap);
+        debug_assert!(src + len <= self.cap, "dst={} src={} len={} cap={}", dst, src, len,
+                      self.cap);
+        ptr::copy_nonoverlapping_memory(
+            self.ptr.offset(dst as int),
+            self.ptr.offset(src as int) as *const T,
```

Comment:
I *think* the `as *const T` should just coerce for free? Still trying to grok when these coercions do or don't kick in, though.